### PR TITLE
3.0rc

### PR DESCRIPTION
**Pull request by jclarkkoala**
# What

Update of namespaces and references to match package naming.

# Why

# How

# CC 🐨

### CODE REVIEW

File: tests/ExplicitModelResolverTest.php [hunk 1]
```diff
@@ -66,7 +66,7 @@ public function testItThrowsModelNotResolvedExceptionIfControllerDoesNotHaveReso
     {
         $route = Mockery::mock(Route::class);
         $route->shouldReceive('getAction')->zeroOrMoreTimes()->andReturn([
-            'uses' => 'Fuzz\MagicBox\Tests\ExplicitExplicitModelResolverTestStubControllerNoResource@getUsers',
+            'uses' => ExplicitExplicitModelResolverTestStubControllerNoResource::class.'@getUsers',
```
Comment (walterbm) [inline tests/ExplicitModelResolverTest.php:69]: Nice! Glad you made this little improvement too!

File: tests/EloquentRepositoryTest.php [hunk 2]
```diff
@@ -46,35 +47,81 @@ public function testItRejectsNonEloquentModels()
...
     {
-        $repo       = $this->getRepository('Fuzz\MagicBox\Tests\Models\User');
+        $repo       = $this->getRepository(User::class);
         $user       = $repo->save();
         $found_user = $repo->find($user->id);
         $this->assertNotNull($found_user);
         $this->assertEquals($user->id, $found_user->id);
     }
 
+    public function testItCanReturnASingleModelFromAQuery()
```
Comment (walterbm) [inline tests/EloquentRepositoryTest.php:69]: Thank you for adding tests!

File: src/Filter.php [hunk 3]
```diff
@@ -35,7 +35,7 @@ class Filter implements QueryFilterContainer
 
     /**
      * Tokens that accept non-scalar filters.
-     * ex: [One,Two,Three,Fuzz]
+     * ex: [One,Two,Three,Koala]
```
Comment (walterbm) [inline src/Filter.php:38]: 🐨

File: src/EloquentRepository.php [hunk 4]
```diff
@@ -635,4 +637,25 @@ private function getRelationsForeignKeyName(HasOneOrMany $relation): string
     {
         return method_exists($relation, 'getForeignKeyName') ? $relation->getForeignKeyName() : $relation->getPlainForeignKey();
     }
+
+    /**
+     * Get the first element against the base query.
+     *
+     * @return \Illuminate\Database\Eloquent\Model
+     */
+    public function first(): ?Model
```
Comment (walterbm) [inline src/EloquentRepository.php:646]: What motivated you to add these methods?

**Review by walterbm — APPROVED** [review]
This looks great and very thorough! 🐨

File: src/EloquentRepository.php (hunk 4 above)
Comment (jclarkkoala) [reply src/EloquentRepository.php:646]: The repository feels incomplete without a function to get the first result from the built-in query.

The only single-result function in the repository is `read($id)` which attempts to query a model by its id, regardless of what query was used to build the repository.

`first` and `firstOrFail` would have been useful when I was using the Repository to power the lookup of a store location by brand_id. Instead I had to modify the repository query to limit it to only one result, like this:
```
$repository->modify()->add(fn (Builder $query) => $query->whereBrandId($brand_id)->limit(1));
$repository->all()->first();
```

**Pull request merged**